models/gyroscope/ethereum/gyroscope_ethereum_trades.sql

```diff
@@ -0,0 +1,29 @@
+{{ config(
+    schema = 'gyroscope_ethereum',
+    tags = ['dunesql'],
+    alias = 'trades',
+    partition_by = ['block_date'],
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['block_date', 'blockchain', 'project', 'version', 'tx_hash', 'evt_index'],
+    post_hook = '{{ expose_spells(\'["ethereum"]\',
+                                "project",
+                                "gyroscope",
+                                \'["fmarrr"]\') }}'
+    )
+}}
+
+{% set project_start_date = '2021-04-20' %}
+
+
+with E_CLPs AS (
+    SELECT
+       pool
+    from {{ source('gyroscope_ethereum','GyroECLPPoolFactory_evt_PoolCreated') }}
+)
+
+SELECT
+*
+FROM ref ('balancer_trades')
+where project_contract_address IN (SELECT pool FROM E_CLPs)
```

Comment:
If you wanted to make your query more efficient you could include another where clause, where the `block_number` >= the min(block_number) from your `E_CLPs` CTE.

Reply:
didn't see how this can be efficient, that would still be including all pools created under GyroECLPPoolFactory. mind to explain please? thanks!

Reply:
Balancer v2 trades contains trades dating back to `2021-04-22`, yet the first ECLP created by `0x412a5b2e7a678471985542757a6855847d4931d5` on Ethereum was at `2023-07-11 21:15`. There is more than 2 years of trades that are going to be used. My understanding is that filtering by `block_number` is more efficient than just address.

I hope you don't mind my comments as I am just trying to help. I'll leave you be now.

Reply:
I see! I was not thinking in this direction, and the final select from balancer v2 trades does not consist block_number so I don't know how to make this work. Does filtering by earliest evt_block_time from E_CLPs CTE would work?

don't get me wrong, I do appreciate your help here! first time doing this so I was a little lost..

Reply:
Awesome! We are always happy to help if you ever have any questions. Any way, your spell on Ethereum could look like this:
`WITH
    gyro_eclp_pools AS (
        SELECT 
            x.pool
            , y.min_block_time
        FROM gyroscope_ethereum.GyroECLPPoolFactory_evt_PoolCreated x
        LEFT JOIN (
            SELECT min(evt_block_time) AS min_block_time 
            FROM gyroscope_ethereum.GyroECLPPoolFactory_evt_PoolCreated 
        ) y
        ON x.pool IS NOT NULL
    )
SELECT * 
FROM balancer_v2_ethereum.trades x
INNER JOIN gyro_eclp_pools y
ON x.block_time >= y.min_block_time
    AND x.project_contract_address = y.pool`

Reply:
Awesome! We are always happy to help if you ever have any questions. Any way, your spell for Ethereum could look like this:
```
WITH
    gyro_eclp_pools AS (
        SELECT 
            x.pool
            , y.min_block_time
        FROM gyroscope_ethereum.GyroECLPPoolFactory_evt_PoolCreated x
        LEFT JOIN (
            SELECT min(evt_block_time) AS min_block_time 
            FROM gyroscope_ethereum.GyroECLPPoolFactory_evt_PoolCreated 
        ) y
        ON x.pool IS NOT NULL
    )
SELECT * 
FROM balancer_v2_ethereum.trades x
INNER JOIN gyro_eclp_pools y
ON x.block_time >= y.min_block_time
    AND x.project_contract_address = y.pool
```

Reply:
Thank you! pushed the commit to reflect the changes here!